nixos/modules/system/activation/top-level.nix

```diff
@@ -363,14 +378,18 @@ in
           fi
         '';
 
-    system.systemBuilderArgs = lib.optionalAttrs (config.system.forbiddenDependenciesRegex != "") {
+    system.systemBuilderArgs = {
+      passedChecks = concatStringsSep " " config.system.checks;
```

Comment:
Where is `passedChecks` use?

Reply:
Nowhere. I've added this comment

```
      # Not actually used in the builder. `passedChecks` is just here to create
      # the build dependencies. Checks are similar to build dependencies in the
      # sense that if they fail, the system build fails. However, checks do not
      # produce any output of value, so they are not used by the system builder.
```

I swear that the text alignment is by accident.

Reply:
And some more
```
      # In fact, using them runs the risk of accidentally adding unneeded paths
      # to the system closure, which defeats the purpose of the `system.checks`
      # option, as opposed to `system.extraDependencies`.
```

Reply:
Awesome as always, Robert!